specs/core/0_beacon-chain.md

```diff
@@ -836,20 +817,19 @@ def update_ancestor_hashes(parent_ancestor_hashes: List[Hash32],
 
 ### Verify attestations
 
-For each `AttestationRecord` object:
+For each `AttestationRecord` object `obj`:
 
 * Verify that `slot <= block.slot - MIN_ATTESTATION_INCLUSION_DELAY` and `slot >= max(parent.slot - CYCLE_LENGTH + 1, 0)`.
-* Verify that `justified_slot` is equal to or earlier than `last_justified_slot`.
+* Verify that `justified_slot` is equal to `justification_source if slot >= block.slot - (block.slot % CYCLE_LENGTH) else prev_cycle_justification_source`
```

Comment:
Replace >= with > ?

Reply:
Don't think that fixes it.
The problem is with the block that triggers the state recalc. It might not even be at the cycle boundary.

We can check it with `last_state_recalculation_slot`

```suggestion
* Verify that `justified_slot` is equal to `justification_source if slot >= state.last_state_recalculation_slot else prev_cycle_justification_source`
```